### PR TITLE
Start of the scanner implementation

### PR DESCRIPTION
**Pull request by alefiury**
This PR refers to an initial implementation of the scanner.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,4 +14,9 @@ This is a compiler for mgol implemented in go for the final work of the compiler
 ## TODO
 
 - [ ] Implement scanner
+- [ ] Implement pattern recognition
+- [ ] Implement scanner tests
 - [x] Check why the readme is a binary file
+- [x] Implement symbol table
+- [x] Implement symbol table tests
```
Comment (MatheusNtg) [inline README.md]: Did you implemented the symbol table on this PR?

File: src/lexer/scanner.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import (
+	"io"
+	"log"
+	"os"
+)
+
+func Scanner(file *os.File) *Token {
+	buffer := make([]byte, 1)
+	nBytes, err := file.Read(buffer)
+
```
Comment (MatheusNtg) [inline src/lexer/scanner.go:13]:
```suggestion

```

**Review by MatheusNtg — CHANGES_REQUESTED** [review]
Overal the PR looks good, just some small alingments to be done and we need to remove the duplicated code regarding to the Scanner function on the main function

File: src/lexer/scanner.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import (
+	"io"
+	"log"
+	"os"
+)
+
+func Scanner(file *os.File) *Token {
+	buffer := make([]byte, 1)
+	nBytes, err := file.Read(buffer)
```
Comment (MatheusNtg) [inline src/lexer/scanner.go]: nBytes is unused

File: src/lexer/scanner.go [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+	if err != nil && err != io.EOF {
+		log.Fatal(err)
+	}
+
+	if err == io.EOF {
+		EofToken := NewToken(EOF, "", NULL)
+		return EofToken
+	}
```
Comment (MatheusNtg) [inline src/lexer/scanner.go:21]: You need to return something on the final of the function

**Review by MatheusNtg — CHANGES_REQUESTED** [review]
Some small updates

File: README.md (hunk 1 above)
Comment (alefiury) [reply README.md]: I did not implement the symbol table on this PR, so i will be removing this modification from the readme file.

**Review by MatheusNtg — APPROVED** [review]

**Pull request merged**